File: crashpad-client/src/interfaces/BookingResponseDTO.ts
export interface BookingResponseDTO {
    bookingId: number;
    startDate: string; // Using string to match the format 'dd-MM-yyyy'
    endDate: string; // Using string to match the format 'dd-MM-yyyy'
    userCreationDate: string;
    userModifyDate: string;
    statusOfBooking: string;
    totalCost: number;
    specialRequests: string;
    hostId: number;
    travelerId: number;

    // Property details
    propertyId: number;
    propertyType: string;
    title: string;
    name: string;
    street: string;
    city: string;
    state: string;
    zip: string;
    capacity: number;
    padMaxLength: string;
    padMaxWidth: string;
    description: string;
    availability: string;
    originalPrice: string;
    discountedPrice: string;

    // Property images
    imageUrls: string[];
}

File: crashpad-server/src/main/java/com/crashpad/springjwt/dto/BookingResponseDTO.java
package com.crashpad.springjwt.dto;

import lombok.Data;

import java.util.List;

@Data
public class BookingResponseDTO {
    private Long bookingId;
    private String startDate;
    private String endDate;
    private String userCreationDate;
    private String userModifyDate;
    private String statusOfBooking;
    private Double totalCost;
    private String specialRequests;
    private Long hostId;
    private Long travelerId;

    // Property details
    private Long propertyId;
    private String propertyType;
    private String title;
    private String name;
    private String street;
    private String city;
    private String state;
    private String zip;
    private Integer capacity;
    private String padMaxLength;
    private String padMaxWidth;
    private String description;
    private String availability;
    private String originalPrice;
    private String discountedPrice;

    // Property images
    private List<String> imageUrls;
}
